src/openloco/windows/constructionwnd.cpp

```diff
@@ -1,75 +1,4836 @@
+
+    static window* nonTrackWindow()
+    {
+        auto window = WindowManager::find(WindowType::construction);
+
+        if (window != nullptr)
+        {
+            setDisabledWidgets(window);
+        }
+
+        window = WindowManager::find(WindowType::construction);
+
+        if (window != nullptr)
+        {
+            window->call_on_mouse_up(common::widx::tab_station);
+        }
+        return window;
+    }
+
+    static window* trackWindow()
+    {
+        auto window = WindowManager::find(WindowType::construction);
+
+        if (window != nullptr)
+        {
+            setDisabledWidgets(window);
+        }
+
+        common::activateSelectedTrackWidgets();
+        window = WindowManager::find(WindowType::construction);
+
+        if (window != nullptr)
+        {
+            window->call_on_mouse_up(construction::widx::rotate_90);
+        }
+        return window;
+    }
+
+    // 0x004A3B0D
+    window* openWithFlags(const uint32_t flags)
+    {
+        auto mainWindow = WindowManager::getMainWindow();
+        if (mainWindow)
+        {
+            auto viewport = mainWindow->viewports[0];
+            _word_1135F86 = viewport->flags;
+        }
+
+        auto window = WindowManager::find(WindowType::construction);
+
+        if (window != nullptr)
+        {
+            if (flags & (1 << 7))
+            {
+                auto trackType = flags & ~(1 << 7);
+                auto roadObj = objectmgr::get<road_object>(trackType);
+
+                if (roadObj->flags & 8)
+                {
+                    if (_trackType & (1 << 7))
+                    {
+                        trackType = _trackType & ~(1 << 7);
+                        roadObj = objectmgr::get<road_object>(trackType);
+
+                        if (roadObj->flags & 8)
+                        {
+                            _trackType = flags;
+
+                            sub_4A3A50();
+
+                            _lastSelectedTrackPiece = 0;
+                            _lastSelectedTrackGradient = 0;
+
+                            return window;
+                        }
+                    }
+                }
+            }
+        }
+
+        WindowManager::closeConstructionWindows();
+        sub_4CD454();
+
+        mainWindow = WindowManager::getMainWindow();
+
+        if (mainWindow)
+        {
+            auto viewport = mainWindow->viewports[0];
+            viewport->flags = _word_1135F86;
+        }
+
+        _trackType = static_cast<uint8_t>(flags);
+        _byte_1136063 = flags >> 24;
+        _x = 0x1800;
+        _y = 0x1800;
+        _word_1135FB8 = 0x100;
+        _constructionRotation = 0;
+        _constructionHover = 0;
+        _byte_113607E = 1;
+        _trackCost = 0x80000000;
+        _byte_1136076 = 0;
+        _lastSelectedTrackPiece = 0;
+        _lastSelectedTrackGradient = 0;
+        _lastSelectedTrackModSection = 0;
+
+        setTrackOptions(flags);
+
+        if (flags & 0x80000000)
```

Comment:
Make this into a split where it goes off to 4 seperate functions: airports, docks, roads, track